### PR TITLE
Use centos 9 base image for dnsmasq

### PR DESCRIPTION
**Pull request by sgoveas**

### CODE REVIEW

**Review by aleskandro — COMMENTED** [review]
/hold

unhold once tests are successful

File: images/bind9/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,7 @@
 FROM quay.io/centos/centos:stream9
 
-RUN dnf -y update && \
-    dnf -y install bind bind-utils && \
-    dnf clean all
+RUN rpm-ostree install bind bind-utils && \
+    ostree container commit
```
[the remark this answers is not in the document]
Comment (sgoveas) [reply images/bind9/Dockerfile:4]: The previous error was 'dnf command not found'. Does this PR https://github.com/openshift/release/pull/43607 need correction to use the correct image? It was set as fcos-bastion.

[the remark this answers is not in the document]
Comment (aleskandro) [reply images/bind9/Dockerfile:4]: oh yes... let's try fixing it here the ones requiring changes tho.

File: images/dhcp/Containerfile [hunk 2]
```diff
@@ -1,5 +1,6 @@
 FROM quay.io/openshifttest/base-alpine:1.2.0
 
-RUN apk add --no-cache dnsmasq
+RUN rpm-ostree install dnsmasq && \
```
Comment (aleskandro) [inline images/dhcp/Containerfile:3]: I see now, you wanted to build dnsmasq on top of base-alpine, but then you replaced the from statement in the testconfigs.

Comment (aleskandro) [reply images/dhcp/Containerfile:3]: Can you change this to build on top of cs9?

File: images/tang/Containerfile [hunk 3]
```diff
@@ -1,8 +1,7 @@
 FROM quay.io/centos/centos:stream8
```
Comment (aleskandro) [inline images/tang/Containerfile:1]:
```suggestion
FROM quay.io/centos/centos:stream9
```

File: images/bind9/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleskandro) [reply images/bind9/Dockerfile:4]: Ok, based on the other comments, this should now be fine.

File: images/dhcp/Containerfile (hunk 2 above)
Comment (aleskandro) [reply images/dhcp/Containerfile:3]: Also adding some comments on the other pr.

File: images/tang/Containerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (aleskandro) [reply images/tang/Containerfile:1]: confirmed cs8 offline.

File: images/dhcp/Containerfile (hunk 2 above)
Comment (sgoveas) [reply images/dhcp/Containerfile]: done

**Pull request merged**